core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php

```diff
@@ -22,43 +22,34 @@
  */
 class MakeServicesPublicPass implements CompilerPassInterface
 {
-    private const SERVICES = [
+    private const IDS = [
         'assets.packages',
+        'database_connection',
         'debug.stopwatch',
         'fragment.handler',
         'lexik_maintenance.driver.factory',
         'monolog.logger.contao',
         'security.authentication.trust_resolver',
+        'security.encoder_factory',
         'security.firewall.map',
-        'security.logout_url_generator',
         'security.helper',
-        'uri_signer',
-    ];
-
-    private const ALIASES = [
-        'database_connection',
+        'security.logout_url_generator',
         'swiftmailer.mailer',
-        'security.encoder_factory',
+        'uri_signer',
     ];
 
     public function process(ContainerBuilder $container): void
     {
-        foreach (self::SERVICES as $service) {
-            if (!$container->hasDefinition($service)) {
-                continue;
+        foreach (self::IDS as $id) {
+            if ($container->hasAlias($id)) {
+                $alias = $container->getAlias($id);
+                $alias->setPublic(true);
             }
 
-            $definition = $container->getDefinition($service);
-            $definition->setPublic(true);
-        }
-
-        foreach (self::ALIASES as $alias) {
-            if (!$container->hasAlias($alias)) {
-                continue;
+            if ($container->hasDefinition($id)) {
```

Comment:
Should this be an `elseif` condition, so we do not look for definitions if the ID is an alias?

Reply:
I don't like `elseif`, if the `$id` is an alias then `$container->hasDefinition($id)` will always evaluate to false.

Reply:
Yes, but if we already know that the ID is an alias, the `hasDefinition()` call is unnecessary, isn't it?

Reply:
Maybe like https://github.com/contao/contao/pull/3759#discussion_r760963494?

Reply:
It's IMO a personal thing. My eyes always hurt a bit if I see `elseif` when not really needed 😅